library/data-model/src/databaseEngine/engine.ts

```diff
@@ -1106,99 +1141,386 @@ class FormOperations {
     }
 
-    // Create all new AVPs (with attachment handling if configured)
-    await this.bulkCreateAvps(avpsToCreate);
+    // Determine if we need a new AVP or can update in-place
+    const shouldCreateNew = await this.shouldCreateNewAvp(
+      currentRevision,
+      parentRevision,
+      fieldname
+    );
 
-    return avpMap;
+    if (shouldCreateNew) {
+      return {type: 'create_new'};
+    } else {
+      return {type: 'update_inplace'};
+    }
   }
 
   /**
-   * Bulk create AVPs, handling attachments if dumper is configured. Uses
-   * PouchDB bulkDocs for efficient batch operations.
+   * Determine if we should create a new AVP or update the existing one in-place.
+   *
+   * We create a new AVP when:
+   * - There's no parent revision (first edit), OR
+   * - The current AVP ID matches the parent's AVP ID (not yet diverged)
    *
-   * NOTE: this operation does not 'dump' attachments - i.e. it does not manage
-   * attachments in any way. The /app layer should use the configured attachment
-   * service.
+   * We update in-place when:
+   * - The current AVP ID differs from parent's (already diverged in a previous edit)
+   */
+  private async shouldCreateNewAvp(
+    currentRevision: ExistingRevisionDBDocument,
+    parentRevision: ExistingRevisionDBDocument | undefined,
+    fieldname: string
+  ): Promise<boolean> {
+    if (!parentRevision) {
+      // No parent - we can update in-place
+      return false;
+    }
+
+    const currentAvpId = currentRevision.avps[fieldname];
+    const parentAvpId = parentRevision.avps[fieldname];
+
+    // If AVP IDs match, this is the first divergence - create new AVP
+    // If AVP IDs differ, we've already diverged - update in-place
+    return currentAvpId === parentAvpId;
+  }
+
+  /**
+   * Apply the determined AVP strategy, updating the appropriate collections.
+   */
+  private async applyAvpStrategy({
+    strategy,
+    fieldname,
+    newFieldData,
+    currentRevision,
+    parentRevision,
+    updatedBy,
+    avpsToCreate,
+    avpsToUpdate,
+    outputAvpMap,
+  }: {
+    strategy: AvpStrategy;
+    fieldname: string;
+    newFieldData: FormDataEntry;
+    currentRevision: ExistingRevisionDBDocument;
+    parentRevision?: ExistingRevisionDBDocument;
+    updatedBy: string;
+    avpsToCreate: Array<NewAvpDBDocument>;
+    avpsToUpdate: Array<ExistingAvpDBDocument>;
+    outputAvpMap: Record<string, string>;
+  }) {
+    switch (strategy.type) {
+      case 'reuse_current':
+        // No changes needed, just reference the existing AVP
+        outputAvpMap[fieldname] = currentRevision.avps[fieldname];
+        break;
+
+      case 'reuse_parent':
+        if (!parentRevision) {
+          throw new Error(
+            'Cannot apply the reuse_parent AVP strategy when parentRevision is not provided.'
+          );
+        }
+        // Reference the parent's AVP (user reverted to previous version)
+        outputAvpMap[fieldname] = parentRevision.avps[fieldname];
```

Comment:
if we're re-using the parent because an edit has changed a value back to a parent, where do we delete the previously created avp?